Sources/SKLogging/CustomLogStringConvertible.swift

```diff
@@ -86,58 +87,51 @@ extension Optional where Wrapped: CustomLogStringConvertible {
   }
 }
 
-/// A JSON-like description of the object.
-public func recursiveDescription(of subject: Any) -> String {
-  return "{"
-    + Mirror(reflecting: subject).children.map { (key, value) in
-      "\(key ?? "<nil>"): \(String(describing: value))"
-    }.joined(separator: ", ")
-    + "}"
-}
-
-fileprivate protocol OptionalProtocol {
-  associatedtype Wrapped
-  var asOptional: Wrapped? { get }
-}
-extension Optional: OptionalProtocol {
-  var asOptional: Wrapped? { self }
-}
-
-/// A JSON-like description of the object that shows trivial values but redacts values that might contain sensitive
-/// information.
-public func recursiveRedactedDescription(of value: Any) -> String {
-  switch value {
-  case let value as Bool:
-    return value.description
-  case let value as Int:
-    return value.description
-  case let value as Double:
-    return value.description
-  case let value as String:
-    return value.hashForLogging
-  case let value as any OptionalProtocol:
-    if let value = value.asOptional {
-      return recursiveRedactedDescription(of: value)
-    } else {
-      return "nil"
+extension Encodable {
+  package var prettyPrintedJSON: String {
+    let encoder = JSONEncoder()
+    encoder.outputFormatting.insert(.prettyPrinted)
+    encoder.outputFormatting.insert(.sortedKeys)
+    guard let data = try? encoder.encode(self) else {
+      return "\(self)"
+    }
+    guard let string = String(data: data, encoding: .utf8) else {
+      return "\(self)"
     }
-  default:
-    break
+    // Don't escape '/'. Most JSON readers don't need it escaped and it makes
+    // paths a lot easier to read and copy-paste.
+    return string.replacingOccurrences(of: "\\/", with: "/")
   }
 
-  let children = Mirror(reflecting: value).children.sorted { $0.label ?? "" < $1.label ?? "" }
-  if !children.isEmpty {
-    let childrenKeyValuePairs = children.map { (label, value) -> String in
-      let label = label ?? "<nil>"
-      let value =
-        if let value = value as? any CustomLogStringConvertible {
-          value.redactedDescription
-        } else {
-          recursiveRedactedDescription(of: value)
-        }
-      return "\(label): \(value)"
+  package var prettyPrintedRedactedJSON: String {
+    func redact(subject: Any) -> Any {
+      if let subject = subject as? [String: Any] {
+        return subject.mapValues { redact(subject: $0) }
```

Comment:
In principle you could have a dictionary keyed by a private value e.g a file path string, I guess that's not a concern currently?

Reply:
Yes, I’m not aware of any requests that store sensitive data in the keys.